images/virtualization-artifact/pkg/controller/vm/internal/lifecycle.go

```diff
@@ -127,83 +127,80 @@ func (h *LifeCycleHandler) syncMigrationState(vm *virtv2.VirtualMachine, kvvm *v
 	} else {
 		vm.Status.MigrationState = h.wrapMigrationState(kvvmi.Status.MigrationState)
 	}
+	setCondition := newSetCondition(vm)
 
-	//nolint:staticcheck
-	mgr := conditions.NewManager(vm.Status.Conditions)
 	cbMigrating := conditions.NewConditionBuilder(vmcondition.TypeMigrating).Generation(vm.GetGeneration())
-	if vm.Status.MigrationState != nil &&
+
+	switch {
+	case vm.Status.MigrationState != nil &&
 		vm.Status.MigrationState.StartTimestamp != nil &&
-		vm.Status.MigrationState.EndTimestamp == nil {
-		mgr.Update(cbMigrating.
-			Status(metav1.ConditionTrue).
-			Reason(vmcondition.ReasonVmIsMigrating).
-			Condition())
-	} else {
-		mgr.Update(cbMigrating.
-			Status(metav1.ConditionFalse).
-			Reason(vmcondition.ReasonVmIsNotMigrating).
-			Condition())
+		vm.Status.MigrationState.EndTimestamp == nil:
+
+		cbMigrating.Status(metav1.ConditionTrue).Reason(vmcondition.ReasonVmIsMigrating)
+		setCondition(cbMigrating)
+
+	case kvvmi != nil && kvvmi.Status.MigrationState != nil &&
+		kvvmi.Status.MigrationState.EndTimestamp != nil &&
+		kvvmi.Status.MigrationState.Failed:
+
+		msg := kvvmi.Status.MigrationState.FailureReason
+		cbMigrating.Status(metav1.ConditionFalse).
+			Reason(vmcondition.ReasonLastMigrationFinishedWithError).
+			Message(msg)
+		setCondition(cbMigrating)
```

Comment:
You don't do return, is it expected? 
Looks like this error can be overridden below

Reply:
this is expected. the syncMigrationState method updates 2 conditions, so I don't return